pkg/relayer/proxy/synchronous.go

```diff
@@ -101,6 +103,28 @@ func (sync *synchronousRPCServer) ServeHTTP(writer http.ResponseWriter, request
 	ctx := request.Context()
 	defer request.Body.Close()
 
+	sync.logger.Debug().Msg("serving synchronous relay request")
+
+	// Extract the relay request from the request body.
+	sync.logger.Debug().Msg("extracting relay request from request body")
+	relayRequest, err := sync.newRelayRequest(request)
+	if err != nil {
+		proxyName := sync.proxyConfig.ProxyName
+		sync.replyWithError(ctx, []byte{}, writer, proxyName, unknownServiceId, err)
```

Comment:
1. Should we not pass the service id of the request even when there's an error?
2. I think even an empty string would cause less confusion in the case of an error

Reply:
1. In that code branch we were not able to parse the `RelayRequest` to extract the serviceId from it.
2. Will pass an empty string :+1: